.dockerignore

```diff
@@ -0,0 +1,43 @@
+# Reminders:
+# * Matching rules are different to `.gitignore`
+# * A pattern without '**' matches in the project's root directory only
+# * Leading and trailing '/' are discarded (it is not possible to
+#   distinguish between files and directories)
+# * More details: https://github.com/balena-io-modules/dockerignore
+
+# development and testing tools or IDEs
+**/*.log
+**/*.pid
+**/*.seed
+.idea
+.lock-wscript
+.nvmrc
+.nyc_output
+.vscode
+coverage
+lib-cov
+logs
+pids
+
+# OS cache files
+**/.DS_Store
+
+# balena CLI config and build files
+**/.balenaconf
+**/.fast-boot.json
+**/.resinconf
+balenarc.yml
+build
+build-bin
+dist
+node_modules
+oclif.manifest.json
+package-lock.json
+resinrc.yml
+tmp
+
+# Docker
+docker-compose*.yml
+.dockerignore
+**/Dockerfile
+**/DOCKER.md
```

Comment:
I don't mind putting them all back in, but I found it really annoying that the entire image would rebuild when I was making minor changes to documentation. Because these files get included in the context, and are copied in the first build stage, adding a character to DOCKER.md invalidated my entire image cache and would rebuild the CLI from source :cry: 

I can remove them from the ignore file now though since we are nearly done testing!

Reply:
Oh right. Eventually we could replace `COPY . .` with "a list of the folders and files that matter" and maybe multiple COPY instructions at strategic positions in the Dockerfiles. A list of files and folders that matter exists in `package.json`, `files` section.